### DISCUSSION TITLE
@types/jest issue

### DISCUSSION THREAD

**Discussion by billnbell** [DefinitelyTyped/DefinitelyTyped#57514] [Issues with a @types package]
warn Package @types/jest has been ignored because it contains invalid configuration. Reason: Package subpath './package.json' is not defined by "exports" in ./node_modules/@types/jest/package.json

"@types/jest": "^27.0.3",

Not sure what this means.

**Comment by jlecordier** [comment]
Jest has built-in typescript definition, you do not need to use the @types package

**Comment by renchap** [comment]
> Jest has built-in typescript definition, you do not need to use the @types package

The Jest documentation still encourages you to use `@types/jest` if you want your tests to be fully typed.

And I am also getting the same warning as the original author.

**Comment by darrylyoung** [comment]
I'm also seeing this warning for the `uuid` package and, at least according to comments in their issues, using `@types/*` is recommended in the same way it's recommended for `jest`, as suggested in the comment above.

```
warn Package @types/jest has been ignored because it contains invalid configuration. Reason: Package subpath './package.json' is not defined by "exports" in /.../node_modules/@types/jest/package.json
warn Package @types/uuid has been ignored because it contains invalid configuration. Reason: Package subpath './package.json' is not defined by "exports" in /.../node_modules/@types/uuid/package.json
```

**Comment by chrisui** [comment]
Downgrading and pinning `@types/jest@27.0.2` resolves for now.

Also commented on the merged pr: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57235#issuecomment-977968465

[a comment by an automated account is not shown]
